src/plugins/visualizations/public/mocks.ts

```diff
@@ -16,22 +16,60 @@
  * specific language governing permissions and limitations
  * under the License.
  */
-import { VisualizationsSetup, VisualizationsStart } from '.';
 
-export type Setup = jest.Mocked<VisualizationsSetup>;
-export type Start = jest.Mocked<VisualizationsStart>;
+import { PluginInitializerContext } from '../../../core/public';
+import { VisualizationsSetup, VisualizationsStart } from './';
+import { VisualizationsPlugin } from './plugin';
+import { coreMock } from '../../../core/public/mocks';
+import { embeddablePluginMock } from '../../../plugins/embeddable/public/mocks';
+import { expressionsPluginMock } from '../../../plugins/expressions/public/mocks';
+import { dataPluginMock } from '../../../plugins/data/public/mocks';
+import { usageCollectionPluginMock } from '../../../plugins/usage_collection/public/mocks';
+import { uiActionsPluginMock } from '../../../plugins/ui_actions/public/mocks';
 
-const createSetupContract = (): Setup => {
-  const setupContract: Setup = undefined;
-  return setupContract;
-};
+const createSetupContract = (): VisualizationsSetup => ({
+  createBaseVisualization: jest.fn(),
+  createReactVisualization: jest.fn(),
+  registerAlias: jest.fn(),
+  hideTypes: jest.fn(),
+});
+
+const createStartContract = (): VisualizationsStart => ({
+  get: jest.fn(),
+  all: jest.fn(),
+  getAliases: jest.fn(),
+  savedVisualizationsLoader: {} as any,
+  showNewVisModal: jest.fn(),
+  createVis: jest.fn(),
+  convertFromSerializedVis: jest.fn(),
+  convertToSerializedVis: jest.fn(),
+});
+
+const createInstance = async () => {
+  const plugin = new VisualizationsPlugin({} as PluginInitializerContext);
+
+  const setup = plugin.setup(coreMock.createSetup(), {
+    data: dataPluginMock.createSetupContract(),
+    expressions: expressionsPluginMock.createSetupContract(),
+    embeddable: embeddablePluginMock.createSetupContract(),
+    usageCollection: usageCollectionPluginMock.createSetupContract(),
+  });
+  const doStart = () =>
+    plugin.start(coreMock.createStart(), {
+      data: dataPluginMock.createStartContract(),
+      expressions: expressionsPluginMock.createStartContract(),
+      uiActions: uiActionsPluginMock.createStartContract(),
+    });
 
-const createStartContract = (): Start => {
-  const startContract: Start = undefined;
-  return startContract;
+  return {
+    plugin,
+    setup,
+    doStart,
+  };
 };
 
-export const expressionsPluginMock = {
+export const visualizationsPluginMock = {
   createSetupContract,
   createStartContract,
+  createInstance,
```

Comment:
I see we added `createInstance`, but is it actually used anywhere?

Reply:
this was there before the move, its just the diff is comparing this against the wrong file.